docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,199 @@
+
+## Considerations
+
+### Validation important and architecturally significant
+
+Based on discussion in the Core WG meeting of 2/3/22, it was decided that without validation and some valid list of actual UoM, the ADR was just adding metadata to the profile and thus did not even rise to the level of "significant" architectural decision.  It was further felt that in order to really provide any value to adopters and to get adherence to their chosen units of measure, EdgeX had to allow for a valid list of units of measure to be specified and be used to check profile units - but in a way that is easy to configure/provide without having to rebuild a service for example.  If the units of measure were defined just in the standard configuration file, it would make it hard to change this list in deployments. 
+
+This new UoM ADR is the result of that discussion.  In general, it specifies, through adopter provided configuration, the exact unit of measures that are allowed for the EdgeX instance and any optional reference (such as a specification) where those units are defined.  It does so through either core metadata configuration (inline) or via an external file making it easier to change.
+
+### Use of SenML
+
+[SenML](https://datatracker.ietf.org/doc/html/rfc8428) was suggested as a specification (currently a proposed standard) from which EdgeX may draw some guidance or inspiration with regard to unit of measure representation in "simple sensor measurements and device parameters."
+
+In fact, SenML defines a simple data model (in JSON, CBOR, XML, EXI) for the exchange of what EdgeX would call readings.  A JSON example is below:
+
+``` json
+[{"n":"urn:dev:ow:10e2073a01080063","u":"Cel","v":23.1}]
+```
+
+In the example above, the array (what EdgeX would consider a collection of readings) has a single SenML Record with a
+measurement for a sensor named "urn:dev:ow:10e2073a01080063" with a current value of 23.1 for degrees measured in Celsius (Cel) unit of measure.  However, SenML suggests the use of short names for the keys in most cases, but long names could be used.  In which case, the JSON SenML reading would look like the following:
+
+``` json
+[{"Name":"urn:dev:ow:10e2073a01080063","Unit":"Cel","Value":23.1}]
+```
+
+In this way, the parallels to EdgeX model are, by accident, uncanny - at least in the JSON instance.  SenML goes to much more depth to provide extensions and more definitions around measurements.  But at its base, the EdgeX format is not unlike SenML and could easily be aligned with SenML in the future (or allow for an application service to export in SenML with an additional function fairly easily and if there were demand).
+
+However, on the basis of "unit of measure", SenML is actually light on details.  This ADR and the proposal here goes much further than what SenML specifies.  With regard to UoM, the SenML specification only says:
```

Comment:
Please also see SenML Units Registry: https://datatracker.ietf.org/doc/html/rfc8428#section-12.1

Reply:
Added and cleaned up this section a bit Farshid.